src/Fixer/ConfigurationDefinitionFixerInterface.php

```diff
@@ -15,6 +15,8 @@
 use PhpCsFixer\OptionsResolver;
 
 /**
+ * Maybe better the other way around, i.e. now there is no upgrade path to the new interface without deprecation warnings
```

Comment:
This is more of concept and I would like to hear what others think.
By deprecating the old way and at the same time deprecating the new way, it leaves no upgrade path without using deprecations. I think we should try avoid that. I think we can remove the annotation here and add later when we have another upgrade path, for example when 3.0 is released or something similar.